ui/app/AppLayouts/Onboarding/views/ProfileFetchingView.qml

```diff
@@ -0,0 +1,118 @@
+            PropertyChanges { target: title; text: Constants.onboarding.profileFetching.titleForSuccess }
+        },
+        State {
+            name: Constants.startupState.profileFetchingError
+            when: root.startupStore.currentStartupState.stateType === Constants.startupState.profileFetchingError
+            PropertyChanges { target: title; text: Constants.onboarding.profileFetching.titleForError }
+            PropertyChanges { target: description; text: Constants.onboarding.profileFetching.descriptionForError }
+            PropertyChanges { target: button; visible: true}
+            PropertyChanges { target: link; visible: true }
+        }
+    ]
+
+    ColumnLayout {
+        anchors.centerIn: parent
+        height: Constants.keycard.general.onboardingHeight
+        spacing: Style.current.bigPadding
+
+        ProfileFetchingAnimation {
+            id: loadingAnimation
+            Layout.alignment: Qt.AlignHCenter
+            state: root.state
+        }
+
+        StatusBaseText {
+            id: title
+            visible: text.length > 0
+            Layout.alignment: Qt.AlignHCenter
+            font.weight: Font.Bold
+            font.pixelSize: Constants.onboarding.profileFetching.titleFontSize
+        }
+
+        StatusBaseText {
+            id: description
+            property string nextText: ""
+            Layout.alignment: Qt.AlignHCenter
+            horizontalAlignment: Text.AlignHCenter
+            visible: text.length > 0
+            wrapMode: Text.WordWrap
+            color: Style.current.secondaryText
+            font.pixelSize: Constants.onboarding.profileFetching.descriptionFontSize
+            Timer {
+                id: nextTextTimer
+                interval: 2500
+                running: description.nextText !== ""
+                onTriggered: { description.text = description.nextText }
+            }
+        }
+
+        StatusButton {
+            id: button
+            visible: false
+            Layout.alignment: Qt.AlignHCenter
+            focus: true
+            text: Constants.onboarding.profileFetching.tryAgainText
+            onClicked: {
+                root.startupStore.doPrimaryAction()
+            }
+        }
+
+        StatusBaseText {
+            id: link
+            visible: false
+            Layout.alignment: Qt.AlignHCenter
+            font.pixelSize: Constants.keycard.general.buttonFontSize
+            text: Constants.onboarding.profileFetching.createNewProfileText
+            color: Theme.palette.primaryColor1
+            MouseArea {
+                anchors.fill: parent
+                cursorShape: Qt.PointingHandCursor
+                hoverEnabled: true
+                onEntered: {
+                    parent.font.underline = true
+                }
+                onExited: {
+                    parent.font.underline = false
```

Comment:
Ugh, this looks a bit clumsy (onEntered/onExited); can you do better with a `HoverHandler` and a declarative approach?

Reply:
HoverHandler won't be enough to replace the mouseArea because we also need on click handler. I've removed the onEntered/onExited and replaced them with a binding on MouseArea.containsMouse